shell/platform/darwin/macos/framework/Source/FlutterEngine.mm

```diff
@@ -428,16 +428,22 @@ - (FlutterCompositor*)createFlutterCompositor {
   // TODO(richardjcai): Add support for creating a FlutterCompositor
   // with a nil _viewController for headless engines.
   // https://github.com/flutter/flutter/issues/71606
-  if (!_viewController) {
+  FlutterViewController* viewController = _viewController;
+  if (!viewController) {
     return nil;
   }
 
   __weak FlutterEngine* weakSelf = self;
 
+  flutter::FlutterCompositor::GetViewCallback get_view_callback =
```

Comment:
Nit: ObjC++ Google style is to match the naming style to the enclosing method, not the variable type. So a C++ type in an ObjC method is named using objCNamingStyle not cpp_naming_style.